File: Microsoft Office Developer Documentation Team/SharePoint 2010 101 Code Samples/49862-SharePoint 2010 101 Code Samples/SharePoint 2010 Developing Workflow Activities/C#/ECM_WorkflowActivity/Features/ExampleWorkflowActivityFeature/ExampleWorkflowActivityFeature.EventReceiver.cs
using System;
using System.Runtime.InteropServices;
using System.Security.Permissions;
using Microsoft.SharePoint;
using Microsoft.SharePoint.Administration;
using Microsoft.SharePoint.Security;

namespace ECM_WorkflowActivity.Features.ExampleWorkflowActivityFeature
{

    [Guid("5e6a4744-0442-47ea-b249-2eb209d5f9ba")]
    public class ExampleWorkflowActivityFeatureEventReceiver : SPFeatureReceiver
    {

        public override void FeatureActivated(SPFeatureReceiverProperties properties)
        {
            //Get the current web application
            SPWebApplication currentWebApp = (SPWebApplication)properties.Feature.Parent;
            
            //Create a config modification object and set its properties
            SPWebConfigModification modification = new SPWebConfigModification();
            modification.Name = "AuthType";
            modification.Owner = "ExampleActivityLibrary";
            modification.Path = "configuration/System.Workflow.ComponentModel.WorkflowCompiler/authorizedTypes";
            modification.Type = SPWebConfigModification.SPWebConfigModificationType.EnsureChildNode;
            modification.Value = "<authorizedType Assembly=\"ExampleActivityLibrary, Version=1.0.0.0, Culture=neutral, PublicKeyToken=3a18e08921ea71f7\" " +
                "Namespace=\"ExampleActivityLibrary\" TypeName=\"*\" Authorized=\"True\" />";

            //Add and apply the modification
            currentWebApp.WebConfigModifications.Add(modification);
            currentWebApp.WebService.ApplyWebConfigModifications();
        }

        //public override void FeatureDeactivating(SPFeatureReceiverProperties properties)
        //{
        //}

    }
}

File: Microsoft Office Developer Documentation Team/SharePoint 2010 101 Code Samples/49862-SharePoint 2010 101 Code Samples/SharePoint 2010 Developing Workflow Activities/VB/ECM_WorkflowActivity/Features/ExampleWorkflowActivityFeature/ExampleWorkflowActivityFeature.EventReceiver.vb
Option Explicit On
Option Strict On

Imports System
Imports System.Runtime.InteropServices
Imports System.Security.Permissions
Imports Microsoft.SharePoint
Imports Microsoft.SharePoint.Administration
Imports Microsoft.SharePoint.Security

<GuidAttribute("db6ea9eb-f71c-45cf-b8c3-1dd06ec007ba")> _
Public Class ExampleWorkflowActivityFeatureEventReceiver 
    Inherits SPFeatureReceiver


    Public Overrides Sub FeatureActivated(ByVal properties As SPFeatureReceiverProperties)
        'Get the current web application
        Dim currentWebApp As SPWebApplication = TryCast(properties.Feature.Parent, SPWebApplication)

        'Create a config modification object and set its properties
        Dim modification As SPWebConfigModification = New SPWebConfigModification()
        modification.Name = "AuthType"
        modification.Owner = "ExampleActivityLibrary"
        modification.Path = "configuration/System.Workflow.ComponentModel.WorkflowCompiler/authorizedTypes"
        modification.Type = SPWebConfigModification.SPWebConfigModificationType.EnsureChildNode
        modification.Value = "<authorizedType Assembly=""ExampleActivityLibrary, Version=1.0.0.0, Culture=neutral, PublicKeyToken=dc326d8658ad3d11"" " + _
            "Namespace=""ExampleActivityLibrary"" TypeName=""*"" Authorized=""True"" />"

        'Add and apply the modification
        currentWebApp.WebConfigModifications.Add(modification)
        currentWebApp.WebService.ApplyWebConfigModifications()

    End Sub


        'Public Overrides Sub FeatureDeactivating(ByVal properties As SPFeatureReceiverProperties)
        'End Sub
End Class
